fe/fe-core/src/main/java/com/starrocks/catalog/MetadataViewer.java

```diff
@@ -276,6 +277,23 @@ private static List<List<String>> getTabletDistribution(String dbName, String tb
         return result;
     }
 
+    private static List<Long> getAllComputeNodeIds() throws DdlException {
+        SystemInfoService infoService = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo();
+        List<Long> allComputeNodeIds = infoService.getBackendIds(false);
+        if (RunMode.isSharedDataMode()) {
+            // check warehouse
+            long warehouseId = ConnectContext.get().getCurrentWarehouseId();
+            List<Long> computeNodeIs =
+                    GlobalStateMgr.getCurrentState().getWarehouseMgr().getAllComputeNodeIds(warehouseId);
+            if (computeNodeIs.isEmpty()) {
+                Warehouse warehouse = GlobalStateMgr.getCurrentState().getWarehouseMgr().getWarehouse(warehouseId);
+                throw new DdlException("no available compute nodes in warehouse " + warehouse.getName());
+            }
+            allComputeNodeIds.addAll(computeNodeIs);
```

Comment:
will contain duplicate ids, if there are backend nodes there in the system, they will be added in line282 and repeated again in line 292.